src/caret_analyze/plot/visualize_lib/bokeh/stacked_bar.py

```diff
@@ -80,91 +78,33 @@ def create_figure(self) -> Figure:
         color_selector = ColorSelectorFactory.create_instance(coloring_rule='unique')
         if self._case == 'best':
             color_selector.get_color()
-        stacked_bar_source = StackedBarSource(target_objects)
-        fig.add_tools(stacked_bar_source.create_hover())
-        stacked_bar_data, x_width_list = \
-            self._get_stacked_bar_data(data, y_labels, self._xaxis_type, x_label)
-
-        vbar_stacks = []
-        for y_label in y_labels:
-            vbar_stack = fig.vbar_stack(
-                [y_label],
-                x='start time',
-                width='x_width_list',
-                source=stacked_bar_source.generate(y_label, stacked_bar_data, x_width_list),
-                color=color_selector.get_color(y_label)
-            )
-            vbar_stacks.append((y_label, vbar_stack))
-
-        legend = Legend(items=vbar_stacks, location='bottom_left')
+        colors = [color_selector.get_color(y_label) for y_label in y_labels]
+
+        source = StackedBarSource(data, y_labels, self._xaxis_type, x_label)
+        # stack bottom -> top to top -> bottom
+        reversed_y_label = list(reversed(y_labels))
+        stacks = fig.vbar_stack(reversed_y_label, x='start time', width='x_width_list',
+                                color=colors, source=source.to_source())
+
+        # add 'label' and 'latency' data to each bar due to display hover
+        for label, stack in zip(reversed_y_label, stacks):
+            stack.data_source.add([label] * source.x_len, 'label')
+            stack.data_source.add(['latency = ' + str(latency)
+                                   for latency in source.data[label]], 'latency')
```

Comment:
Moving this process to StackedBarSource.to_source() improves computational complexity and module coupling.

Reply:
Fixed it in [7d58270](https://github.com/tier4/CARET_analyze/pull/307/commits/7d582702f850e8001e84b9cb31345618c675ba6f).